material_studies/rally/lib/home.dart

```diff
@@ -0,0 +1,204 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+//import 'package:rally/tabs/accounts.dart';
+//import 'package:rally/tabs/bills.dart';
+//import 'package:rally/tabs/budgets.dart';
+//import 'package:rally/tabs/overview.dart';
+import 'package:rally/tabs/settings.dart';
+
+const int tabCount = 5;
+
+class HomePage extends StatefulWidget {
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage>
+    with SingleTickerProviderStateMixin {
+  TabController _tabController;
+
+  _HomePageState() {
+    _tabController = TabController(length: tabCount, vsync: this);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    print('_HomePageState initState');
+
+    _tabController.addListener(() {
+      if (_tabController.indexIsChanging &&
+          _tabController.previousIndex != _tabController.index) {
+        setState(() {});
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+
+    return Scaffold(
+        body: SafeArea(
+          child: Column(
+            children: <Widget>[
+              Theme(
+                // This theme effectively removes the default visual touch
+                // feedback for tapping a tab, which is replaced with a custom
+                // animation.
+                data: theme.copyWith(
+                  splashColor: Colors.transparent,
+                  highlightColor: Colors.transparent,
+                ),
+                child: TabBar(
+                  // Setting isScrollable to true prevents the tabs from being
+                  // wrapped in [Expanded] widgets, which allows for more
+                  // flexible sizes and size animations among tabs.
+                  isScrollable: true,
+                  labelPadding: EdgeInsets.zero,
+                  tabs: _buildTabs(theme),
+                  controller: _tabController,
+                  // This removes the tab indicator.
+                  indicatorColor: Colors.transparent,
+                ),
+              ),
+              Expanded(
+                child: TabBarView(
+                  controller: _tabController,
+                  children: _buildTabViews(),
+                ),
+              )
+            ],
+          ),
+        ));
+  }
+
+  List<Widget> _buildTabs(ThemeData theme) {
+    return <Widget>[
+      _buildTab(theme, Icons.pie_chart, "OVERVIEW", 0),
+      _buildTab(theme, Icons.attach_money, "ACCOUNTS", 1),
+      _buildTab(theme, Icons.money_off, "BILLS", 2),
+      _buildTab(theme, Icons.table_chart, "BUDGETS", 3),
+      _buildTab(theme, Icons.settings, "SETTINGS", 4),
+    ];
+  }
+
+  List<Widget> _buildTabViews() {
+    return <Widget>[
+//      OverviewView(),
```

Comment:
Add comment here that these views will be added as they are ready

Reply:
Done